pkg/backingstore/backingstore.go

```diff
@@ -530,6 +545,66 @@ func RunDelete(cmd *cobra.Command, args []string) {
 	}
 }
 
+// RunRemovePendingPods runs a CLI command
+func RunRemovePendingPods(cmd *cobra.Command, args []string) {
+	log := util.Logger()
+
+	if len(args) != 1 || args[0] == "" {
+		log.Fatalf(`❌ Missing expected arguments: <backing-store-name> %s`, cmd.UsageString())
+	}
+
+	o := util.KubeObject(bundle.File_deploy_crds_noobaa_io_v1alpha1_backingstore_cr_yaml)
+	backStore := o.(*nbv1.BackingStore)
+	backStore.Name = args[0]
+	backStore.Namespace = options.Namespace
+	backStore.Spec = nbv1.BackingStoreSpec{}
+	if !util.KubeCheck(backStore) {
+		log.Fatalf(`❌ Could not get BackingStore %q in namespace %q`,
+			backStore.Name, backStore.Namespace)
+	}
+	if backStore.Spec.Type != nbv1.StoreTypePVPool {
+		log.Fatalf(`❌ Could not get Run this Command on None PV-Pool backingstore`)
+	}
+
+	nbClient := system.GetNBClient()
+	hostsInfo, err := nbClient.ListHostsAPI(nb.ListHostsParams{Query: nb.ListHostsQuery{Pools: []string{backStore.Name}}})
+	if err != nil {
+		rpcErr, isRPCErr := err.(*nb.RPCError)
+		if !isRPCErr || rpcErr.RPCCode != "NO_SUCH_POOL" {
+			log.Fatalf(`❌ Failed to read BackingStore host info: %s`, err)
+		}
+	}
+	podsList := &corev1.PodList{}
+	util.KubeList(podsList, client.InNamespace(options.Namespace), client.MatchingLabels{"pool": backStore.Name})
+	for _, pod := range podsList.Items {
+		if !isPodinNoobaa(&pod, &hostsInfo.Hosts) {
```

Comment:
Maybe I'm being too strict but there is a theoretical race condition here. 

between the time of `listsHosts` and `kubeDelete(&pod)` the agent might already be connected.

a possible solution can be to recheck the hosts after the pod is deleted and before the PVC is deleted. if the pod became registered in noobaa then you can just deploy the pod again.

it might not be critical to fix right now but we should at least have a technical debt regarding this.

Reply:
We want this command to be running on stable system. User asks for 10 see that he got 3 and can't get more. he will change the NumVolumes to what he see as the max. and just to cleanup after himself he runs this command. This command will be removed as soon as we have scale down.

Reply:
you can never guarantee a stable system. for example some cpu\mem can become available (an unrelated pod got evicted) and a pod that was not attached becomes attached